File: NoteS/services/CommentEditService.cs
using NoteS.exceptions;
using NoteS.models.dto;
using NoteS.models.entity;
using NoteS.models.mappers;
using NoteS.repositories;

namespace NoteS.services;

public class CommentEditService(
    NoteInformationService noteInformationService,
    INoteRepository repository,
    AccountInformationService accountInformationService)
{
    public Task<Note> EditContentComment(NotePathDto pathComment, AccNameDto owner,
        CommentEditRequestDto requestDto)
    {
        var comment = noteInformationService.Get(pathComment, owner);
        if (comment.Type != NoteTypes.Comment) throw new NoteTypeException("комментарий");
        if (!comment.IsEditable()) throw new TimeMissed("редактирования комментариев");
        comment.SyntaxType = requestDto.SyntaxType;
        comment.Content = requestDto.Content;
        comment.Title = requestDto.Title;
        return repository.Save(comment);
    }

    public Task<Note> EditContentComment(NotePathDto pathComment, CommentEditRequestDto requestDto)
    {
        var comment = noteInformationService.Get(pathComment);
        comment.SyntaxType = requestDto.SyntaxType;
        comment.Content = requestDto.Content;
        comment.Title = requestDto.Title;
        return repository.Save(comment);
    }

    public Task<Note> CreateComment(AccNameDto accountName, NotePathDto pathNote,
        CommentCreateRequestDto requestDto)
    {
        var note = noteInformationService.Get(pathNote);
        AccIdDto account = accountInformationService.Get(accountName);
        var comment = new Note(requestDto.Title)
        {
            Path =  Guid.NewGuid().ToString(),
            MainNote = note.Id,
            Content = requestDto.Content,
            Owner = account.Id,
            Type = NoteTypes.Comment,
            IsPublic = true,
            SyntaxType = requestDto.SyntaxType
        };
        return repository.Save(comment);
    }

    public async Task Delete(NotePathDto pathComment, AccNameDto owner)
    {
        var note = noteInformationService.Get(pathComment, owner);
        if (!note.IsEditable()) throw new TimeMissed("удаления комментариев");
        if (!await repository.Delete(note)) throw new DontDel("Комментарий");
    }
}

File: javaBack/ResourceServer/src/main/java/com/notes/services/managers/CommentEditService.java
package com.notes.services.managers;

import com.notes.exceptions.CommentEditTimeMissedException;
import com.notes.exceptions.NoteNotFoundException;
import com.notes.exceptions.NoteTypeException;
import com.notes.mappers.repository.NoteRepositoryMapper;
import com.notes.models.domain.CommentEditDto;
import com.notes.models.domain.NoteContentDomainDto;
import com.notes.models.domain.NoteCreateDto;
import com.notes.models.domain.NoteTypeDomainDto;
import com.notes.models.entity.Account;
import com.notes.models.entity.Note;
import com.notes.models.entity.NoteContent;
import com.notes.models.entity.NoteDto;
import com.notes.repository.NoteRepository;
import com.notes.services.utils.GeneratorUtils;
import com.notes.services.utils.NoteUtils;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;

@Service
@RequiredArgsConstructor
public class CommentEditService {
   private final NoteInformationService noteInformationService;
   private final NoteUtils noteUtils;
   private final NoteRepository noteRepository;
   private final NoteRepositoryMapper noteRepositoryMapper;
   private final GeneratorUtils generatorUtils;
   private final AccountInformationService accountInformationService;

   public NoteContentDomainDto editComment(String pathComment, String ownerName,
                                           CommentEditDto dto) {
      var commentEntity = getComment(pathComment, ownerName);
      commentEntity.content().setSyntaxType(noteRepositoryMapper.of(dto.syntaxType()));
      commentEntity.content().setContent(dto.content());
      commentEntity.note().setNoteType(noteRepositoryMapper.of(NoteTypeDomainDto.COMMENT_REDACTED));
      commentEntity.note().setTitle(dto.title());
      var r = noteRepository.save(commentEntity);
      return noteRepositoryMapper.of(r);
   }

   public NoteContentDomainDto createComment(String accountName, String pathNote,
                                             NoteCreateDto dto) {
      var account = accountInformationService.findAccount(accountName);
      var note = noteInformationService.findPublicByPath(pathNote, accountName);
      var noteNew = Note.builder()
              .path(generatorUtils.generateUUID().toString())
              .mainNote(Note.builder().id(note.id()).build())
              .description(dto.description())
              .title(dto.title())
              .owner(Account.builder().id(account.id()).build())
              .noteType(noteRepositoryMapper.of(NoteTypeDomainDto.COMMENT))
              .syntaxType(noteRepositoryMapper.of(dto.syntaxType()))
              .isPublic(true)
              .build();
      var comment = new NoteDto(noteNew,
              NoteContent.builder()
                      .content(dto.content())
                      .title(dto.title())
                      .title(noteNew.getTitle())
                      .uuid(noteNew.getElasticUuid())
                      .owner(account.id())
                      .noteType(noteNew.getNoteType())
                      .syntaxType(noteNew.getSyntaxType())
                      .build()
      );
      var r = noteRepository.save(comment);
      return noteRepositoryMapper.of(r);
   }

   public void deleteComment(String pathComment, String ownerName) {
      var commentEntity = getComment(pathComment, ownerName);
      noteRepository.delete(commentEntity.note());
   }

   private NoteDto getComment(String pathComment, String ownerName) {
      var comment = noteInformationService.findPublicByPath(pathComment, ownerName);
      var commentEntity = noteRepository.findById(comment.id()).orElseThrow(NoteNotFoundException::new);
      if (!noteUtils.isComment(comment.noteType())) throw new NoteTypeException();
      if (!noteUtils.isEdit(comment)) throw new CommentEditTimeMissedException();
      return commentEntity;
   }
}
